### PR TITLE
Move out forward model

### LINKED ISSUE #4180: Simplify ExtJobList

### PR DESCRIPTION
**Pull request by eivindjahren**
**Issue**
Partly resolves #4180

**Approach**
_Short description of the approach_

## Pre review checklist

- [ ] Added appropriate release note label
- [ ] PR title captures the intent of the changes, and is fitting for release notes.
- [ ] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).

Adding labels helps the maintainers when writing release notes. This is the [list of release note labels](https://github.com/equinor/ert/labels?q=release-notes).

### CODE REVIEW

File: tests/unit_tests/c_wrappers/res/enkf/test_programmatic_res_config.py [hunk 1]
```diff
@@ -273,10 +273,6 @@ def test_large_config(setup_case):
         == prog_model_config.get_history_source()
     )
     assert loaded_model_config.obs_config_file == prog_model_config.obs_config_file
-    assert (
```
Comment (andreas-el) [inline tests/unit_tests/c_wrappers/res/enkf/test_programmatic_res_config.py:276]: assert loaded_res_config.model_config.job_name_list() == prog_res_config.model_config.job_name_list()

Comment (andreas-el) [reply tests/unit_tests/c_wrappers/res/enkf/test_programmatic_res_config.py:276]: Resolved.

File: src/clib/lib/enkf/model_config.cpp [hunk 2]
```diff
@@ -238,22 +232,19 @@ model_config_type *model_config_alloc_empty() {
...
 model_config_type *model_config_alloc(const config_content_type *config_content,
                                       const char *data_root,
-                                      const ext_joblist_type *joblist,
                                       const ecl_sum_type *refcase) {
     model_config_type *model_config = model_config_alloc_empty();
 
     if (config_content)
-        model_config_init(model_config, config_content, data_root, 0, joblist,
-                          refcase);
+        model_config_init(model_config, config_content, data_root, 0, refcase);
```
Comment (andreas-el) [inline src/clib/lib/enkf/model_config.cpp:239]: int ens_size is always zero, and isn't used.
Should we remove?

Comment (eivindjahren) [reply src/clib/lib/enkf/model_config.cpp:239]: Perhaps separate issue, but I can look into it.

Comment (eivindjahren) [reply src/clib/lib/enkf/model_config.cpp:239]: It is unused in the `model_config_init`.

Comment (eivindjahren) [reply src/clib/lib/enkf/model_config.cpp:239]: https://github.com/equinor/ert/pull/4205

**Review by andreas-el — APPROVED** [review]
Approved. :suspect: BR. Doomguy

**Pull request merged**